pkg/controller/s3/bucket/loggingConfig.go

```diff
@@ -167,3 +130,50 @@ func (in *LoggingConfigurationClient) CreateOrUpdate(ctx context.Context, bucket
 func (*LoggingConfigurationClient) Delete(_ context.Context, _ *v1beta1.Bucket) error {
 	return nil
 }
+
+// LateInitialize is responsible for initializing the resource based on the external value
+func (in *LoggingConfigurationClient) LateInitialize(ctx context.Context, bucket *v1beta1.Bucket) error {
+	external, err := in.client.GetBucketLoggingRequest(&awss3.GetBucketLoggingInput{Bucket: awsclient.String(meta.GetExternalName(bucket))}).Send(ctx)
+	if err != nil {
+		return awsclient.Wrap(err, loggingGetFailed)
+	}
+	config := bucket.Spec.ForProvider.LoggingConfiguration
+	if external.LoggingEnabled == nil {
+		// There is no value send by AWS to initialize
+		return nil
+	}
+
+	in.logger.Debug(fmt.Sprintf("called LateInitialize for %s", reflect.TypeOf(in).Elem().Name()))
+
+	if config == nil {
+		// We need the configuration to exist so we can initialize
+		bucket.Spec.ForProvider.LoggingConfiguration = &v1beta1.LoggingConfiguration{}
+		config = bucket.Spec.ForProvider.LoggingConfiguration
+	}
+	// Late initialize the target Bucket and target prefix
+	config.TargetBucket = awsclient.LateInitializeStringPtr(config.TargetBucket, external.LoggingEnabled.TargetBucket)
+	config.TargetPrefix = awsclient.LateInitializeString(config.TargetPrefix, external.LoggingEnabled.TargetPrefix)
+	// If the there is an external target grant list, and the local one does not exist
+	// we create the target grant list
+	if external.LoggingEnabled.TargetGrants != nil && len(config.TargetGrants) == 0 {
```

Comment:
```suggestion
	if len(external.LoggingEnabled.TargetGrants) != 0 && len(config.TargetGrants) == 0 {
```

Length check is always safer than nil check for arrays since array with 0 elements is not nil but empty for us.